contrib/ci/check-headers.py

```diff
@@ -31,16 +34,40 @@ def test_files() -> int:
 
     toplevel_headers = ["libfwupd/fwupd.h", "libfwupdplugin/fwupdplugin.h"]
     toplevel_headers_nopath = [os.path.basename(fn) for fn in toplevel_headers]
+    lib_headers = glob.glob('libfwupd*/*.h')
+    lib_headers.remove ('libfwupd/fwupd.h')
+    lib_headers.remove ('libfwupdplugin/fwupdplugin.h')
+    lib_headers_nopath = [os.path.basename(fn) for fn in lib_headers]
+
+    # test all C and H files
+    for fn in glob.glob("**/*.[c|h]", recursive=True):
+        includes = __get_includes(fn)
+
+        if fn.startswith('plugins'):
```

Comment:
maybe something like `and not fn.endswith('self-test.c')` would reduce the amount of annotations required?